thinc/model.py

```diff
@@ -363,6 +363,22 @@ def walk(self) -> Iterable["Model"]:
             yield node
             queue.extend(node.layers)
 
+    def replace_layer(self, old: "Model", new: "Model") -> "Model":
+        """Replace one of the model's immediate children with a new layer. If the
+        layer is not found within the children, an IndexError is raised. Returns
+        the model."""
+
+        if not old in self.layers:
+            raise IndexError("Cannot replace non-existing layer")
+
+        self._layers = [new if layer == old else layer for layer in self.layers]
+
+        for (name, item) in self._refs.items():
```

Comment:
I was going to say "Oh, we should walk the tree and replace the reference", but then I realised you can't have a reference higher up in the tree, only further down.